src/blocks/post-carousel/inspector.js

```diff
@@ -34,6 +34,25 @@ class Inspector extends Component {
 			columns,
 		} = attributes;
 
+		const columnCountByPosts = ( selectedColumns ) => {
+			const { postsToShow } = attributes;
+			setAttributes( { columns:
+				( selectedColumns > postsToShow ) ? postsToShow : selectedColumns,
+			} );
+		};
```

Comment:
This function should be named in a way that describes what it does. Right now it implies that it will get the column count when in fact, this is setting the column count based on the "posts to show" value. Making this `setColumnCountByPosts` is more descriptive.